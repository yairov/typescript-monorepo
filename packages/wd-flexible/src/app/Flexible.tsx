import React from 'react';
import calculate from 'wd-common/utils/calculate';
import CommonComponent from 'wd-common/components/CommonComponent';

export default function Flexible() {
  return (
    <div>
      <h1>Hey! I'm the "Flexible" project</h1>
      <p>
        I can be a dependency of "Main" project or ran independently.
        <br/><br/>
        We both use the "Common" project.
      </p>
      <CommonComponent text="CommonComponentInFlexible"/>
      <div>
        Calculate using string: {calculate('str')}
      </div>
    </div>
  );
}
