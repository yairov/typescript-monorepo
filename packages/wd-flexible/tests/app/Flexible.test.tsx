import React from 'react';
import {render} from '@testing-library/react';

import Flexible from 'wd-flexible/src/app/Flexible';

describe('Flexible', () => {
  test('simple', () => {
    const {asFragment} = render(<Flexible/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
