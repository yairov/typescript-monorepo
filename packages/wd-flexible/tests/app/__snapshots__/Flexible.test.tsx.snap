// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flexible simple 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hey! I'm the "Flexible" project
    </h1>
    <p>
      I can be a dependency of "Main" project or ran independently.
      <br />
      <br />
      We both use the "Common" project.
    </p>
    <div>
      hey! I'm a div that's being used in both projects! CommonComponentInFlexible
    </div>
    <div>
      Calculate using string: str1
    </div>
  </div>
</DocumentFragment>
`;
